docs/100_OVERVIEW.md

```diff
@@ -0,0 +1,147 @@
+# OCR2 Automation Design Overview
+The intent of the plugin design is to separate responsibility of components,
+ensure modularity, and provide accessible types for viewing content in 
+messages via external sources (like telemetry).
+
+## Glossary of Terms
+- Observation: two different types; one from the `Plugin` and one from an `Observer`
+- Report: encoded instructions and/or values for on-chain execution
+- BlockKey: a block identifier (typically a big.Int on many chains)
```

Comment:
nit. for now we can safely assume it will be uint64 on all chains until a new case comes up